pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,345 @@ import (
 type StoragePool struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec is the spec for a StoragePool resource
+	Spec StoragePoolSpec `json:"spec,omitempty"`
+	// Status is for handling status of StoragePool resource
+	Status StoragePoolStatus `json:"status,omitempty"`
 }
 
+// StoragePoolSpec is the spec for a StoragePool resource
+type StoragePoolSpec struct {
+	// StorageCohortName of the cohort the pool is a part of
+	StorageCohortName string `json:"storageCohortName,omitempty"`
+
+	// PoolCofiguration for the configuration related to pool
+	PoolCofiguration map[string]string `json:"poolCofiguration,omitempty"`
+
+	// DeviceSpec for device related filtering, configuration
+	DeviceSpec DeviceSpec `json:"deviceSpec,omitempty"`
+
+	// Capabilities of the Storage pool and the supported configurations
+	Capabilities Capabilities `json:"capabilities,omitempty"`
+
+	// RequestedCapacity for the pool creation
+	RequestedCapacity resource.Quantity `json:"requestedCapacity,omitempty"`
+}
+
+// StoragePoolStatus is for handling status of StoragePool resource
+type StoragePoolStatus struct {
+	// ReferenceResource points to the pre-existing pool resource
+	ReferenceResource ReferenceResource `json:"referenceResource,omitempty"`
+
+	// StorageCapacity of the pool,viz total, used and available capacity
+	StorageCapacity StorageCapacity `json:"storageCapacity,omitempty"`
+
+	// StorageIOPs of the pool,viz total, provisioned, used and available capacity
+	StorageIOPs StorageIOPs `json:"storageIops,omitempty"`
+
+	// DeviceConfiguration for list of Device UIDs and raid configurations currently active
+	DeviceConfiguration DeviceConfiguration `json:"deviceConfiguration,omitempty"`
+
+	//VolumeSizeMaxLimit for maximum volume size allowed
+	VolumeSizeMaxLimit resource.Quantity `json:"volumeSizeMaxLimit,omitempty"`
+
+	// State of the pool based on various scenarios
+	State StoragePoolState `json:"state,omitempty"`
```

Comment:
Should we have phase and conditions as in strage-cohort.go?
cc @shovanmaity

Reply:
We can have.. Trying to list what information we want to put here ...

1. If pool has low available capacity it can be here
2. If pool has bad disk or any error reported for the pool it can be here.

Reply:
Adding Phase and Conditions, removing State.